js/chatbot/shared/ChatBot.svelte

```diff
@@ -337,6 +337,9 @@
 					{allow_file_downloads}
 					on:copy={(e) => dispatch("copy", e.detail)}
 				/>
+				{#if generating && messages[messages.length - 1].role === "assistant" &&messages[messages.length - 1].metadata.status === "done"}
```

Comment:
we may need to check that messages.length>0 and that the metadata and status props are present?
```suggestion
				{#if generating && messages[messages.length - 1].role === "assistant" &&messages[messages.length - 1].metadata?.status? === "done"}
```